In [1]:
import pandas as pd 

In [10]:
pat = [5,4,4,3,9,4]
jack = [4,8,7,5,1,5]
alex = [9,9,8,10,4,10]

In [11]:
all_scores = pat + jack + alex
print(all_scores)

[5, 4, 4, 3, 9, 4, 4, 8, 7, 5, 1, 5, 9, 9, 8, 10, 4, 10]


In [12]:
data = pd.DataFrame({
    'Person': (['Pat']*len(pat)) + (['Jack']*len(jack)) + (['Alex']*len(alex)),
    'Score' : all_scores
})
data

Person  Score
0     Pat      5
1     Pat      4
2     Pat      4
3     Pat      3
4     Pat      9
5     Pat      4
6    Jack      4
7    Jack      8
8    Jack      7
9    Jack      5
10   Jack      1
11   Jack      5
12   Alex      9
13   Alex      9
14   Alex      8
15   Alex     10
16   Alex      4
17   Alex     10

In [13]:
data.groupby('Person').mean()

Score
Person          
Alex    8.333333
Jack    5.000000
Pat     4.833333

In [16]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [17]:
lm = ols('Score ~ Person', data = data).fit()

In [18]:
table = sm.stats.anova_lm(lm)
print(table)

            df     sum_sq    mean_sq         F    PR(>F)
Person     2.0  46.777778  23.388889  4.488273  0.029668
Residual  15.0  78.166667   5.211111       NaN       NaN


#### H0: The means of all populations are equal

##### Since the p_value = 0.029668 is less than alpha = 0.05 , we reject the Null Hypothesis i.e. We reject that the means of all populations are equal.

#### Therefore, according to the means, Alex is the best Archer as he has the highest average scoring